src/modules/order/components/PaymentInput/index.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, InputHTMLAttributes } from 'react';
+import { RegisterOptions, UseFormRegisterReturn } from 'react-hook-form';
+import { paymentForm } from '../../conteiners/PaymentDetail';
+import styles from './index.module.scss';
+
+interface FormErrors {
+  [key: string]: {
+    message?: string;
+  };
+}
+
+interface PaymentInputProps extends InputHTMLAttributes<HTMLInputElement> {
+  type?: string;
+  className?: string;
+  placeholder?: string;
+  rules?: RegisterOptions;
+  id: string;
+  name: keyof paymentForm;
+  errors: FormErrors;
+  register: (
+    name: keyof paymentForm,
+    options?: RegisterOptions,
+  ) => UseFormRegisterReturn;
+  onChage?: () => void;
+}
+
+const PaymentInput: FC<PaymentInputProps> = ({
+  id,
```

Comment:
Not just input. It works together with a reack-hook-form. Our custom input does not work with a register. We can create something like "InputHookForm" in a core  to avoid breaking the existing input. What do you think?

Reply:
I think it's better to improve the existing input component by integrating react-hook-form and making it controlled. If you think a specific component is needed, that's fine. But then, I think you should implement validation and handling inside the component.